src/aosm/azext_aosm/tests/latest/test_aosm_vnf_publish_and_delete.py

```diff
@@ -105,16 +105,30 @@ def test_vnf_nsd_publish_and_delete(self, resource_group):
                     f'az aosm nfd publish -f "{nfd_input_file_path}" --definition-type vnf'
                 )
                 break
-            except Exception:
+            except Exception as exception:
                 retry_attempts += 1
 
+                if retry_attempts == 2:
+                    # If the command fails twice, then the test should fail.
+                    # We still need to clean up the resources, so we run the delete command.
+                    self.cmd(
+                        f'az aosm nfd delete --definition-type vnf -f "{nfd_input_file_path}" --clean --force'
+                    )
+                    raise exception
+
         nsd_input_file_path = update_resource_group_in_input_file(
             NSD_INPUT_TEMPLATE_NAME, NSD_INPUT_FILE_NAME, resource_group
         )
 
         self.cmd(f'az aosm nsd build -f "{nsd_input_file_path}" --force')
 
-        self.cmd(f'az aosm nsd publish -f "{nsd_input_file_path}"')
+        try:
+            self.cmd(f'az aosm nsd publish -f "{nsd_input_file_path}"')
+        except Exception as exception:
```

Comment:
The normal way to handle this is with a finally block rather than an except block.

I.e.
```
try:
    publish_command
finally:
    delete_command
```

Then the delete will always run and if there is an exception from the publish command it will be raised afterwards.